File: annotation-processor/src/test/java/me/bristermitten/mittenlib/annotations/parser/CompilerMessagesTest.java
package me.bristermitten.mittenlib.annotations.parser;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class CompilerMessagesTest {

    @Test
    void testConfigClassParserCompilerMessages() {
        ConfigClassParserCompilerMessages[] values = ConfigClassParserCompilerMessages.values();
        assertTrue(values.length > 0);

        ConfigClassParserCompilerMessages val = ConfigClassParserCompilerMessages.valueOf(values[0].name());
        assertNotNull(val);
    }

    @Test
    void testCustomDeserializersCompilerMessages() {
        CustomDeserializersCompilerMessages[] values = CustomDeserializersCompilerMessages.values();
        assertTrue(values.length > 0);

        CustomDeserializersCompilerMessages val = CustomDeserializersCompilerMessages.valueOf(values[0].name());
        assertNotNull(val);
    }
}

File: annotation-processor/src/test/scala/me/bristermitten/mittenlib/annotations/parser/CompilerMessagesTest.scala
package me.bristermitten.mittenlib.annotations.parser

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class CompilerMessagesTest extends AnyFunSuite with Matchers {

  test("testCustomDeserializersCompilerMessages") {
    val values = CustomDeserializersMessagesCompilerMessages.values()
    values.length should be > 0

    val valName =
      CustomDeserializersMessagesCompilerMessages.valueOf(values(0).name())
    valName should not be null
  }
}
